src/main/java/com/gluonhq/substrate/target/IosTargetConfiguration.java

```diff
@@ -88,12 +107,66 @@ public String getAdditionalSourceFileLocation() {
         return Arrays.asList(objectFile.toAbsolutePath().toString());
     }
 
+    @Override
+    public boolean link(ProcessPaths paths, ProjectConfiguration projectConfiguration) throws IOException, InterruptedException {
+        boolean result = super.link(paths, projectConfiguration);
+
+        if (result) {
+            createInfoPlist(paths, projectConfiguration);
+
+            if (! isSimulator()) {
+                CodeSigning codeSigning = new CodeSigning(paths, projectConfiguration);
+                if (! codeSigning.signApp()) {
+                    Logger.logSevere("Error signing the app");
+                }
+            }
+        }
+        return result;
+    }
+
     @Override
     public String getCompiler() {
         return "clang";
     }
 
+    @Override
+    String getAppPath(String appName) {
+        Path appPath = paths.getAppPath().resolve(appName + ".app");
+        if (!Files.exists(appPath)) {
+            try {
+                Files.createDirectories(appPath);
+            } catch (IOException e) {
+                Logger.logSevere("Error creating path " + appPath + ": " + e.getMessage());
+            }
+        }
+        return appPath.toString() + "/" + appName;
+    }
+
+    private String getArch() {
+        return projectConfiguration.getTargetTriplet().getArch();
+    }
+
     private String getSysroot() {
-        return sysroot;
+        return isSimulator() ? XcodeUtils.SDKS.IPHONESIMULATOR.getSDKPath() :
+                XcodeUtils.SDKS.IPHONEOS.getSDKPath();
+    }
+
+    private boolean isSimulator() {
+        return projectConfiguration.getTargetTriplet().getProfile().equals(Constants.Profile.IOS_SIM);
+    }
+
+    private void createInfoPlist(ProcessPaths paths, ProjectConfiguration projectConfiguration) {
+        try {
+            InfoPlist infoPlist = new InfoPlist(paths, projectConfiguration, isSimulator() ?
+                    XcodeUtils.SDKS.IPHONESIMULATOR : XcodeUtils.SDKS.IPHONEOS);
+            Path plist = infoPlist.processInfoPlist();
+            if (plist != null) {
```

Comment:
is plist allowed to be null ?

Reply:
It will fail before (with RTE), but since the method returns `null`, this prevents the warning of possible null pointer exception.